netZooPy/panda/panda.py

```diff
@@ -450,9 +450,8 @@ def processData(
         ):
             # Initialize data & Populate gene expression
             self.expression = np.zeros((self.num_genes, self.expression_data.shape[1]))
-            idx_geneEx = [gene2idx.get(x, 0) for x in self.expression_genes]
-            
-            self.expression[idx_geneEx, :] = self.expression_data.values
+            idx_geneEx = [gene2idx.get(x) for x in self.gene_names]
+            self.expression[idx_geneEx, :] = self.expression_data.loc[self.gene_names].values
```

Comment:
@michelegentili93 what happens here (self.expression_data.loc[self.gene_names]) in case of modeprocess='union'? I think we might get an error because gene names will be union and some genes may not exist in expression coming from motif.

Reply:
> @michelegentili93 what happens here (self.expression_data.loc[self.gene_names]) in case of modeprocess='union'? I think we might get an error because gene names will be union and some genes may not exist in expression coming from motif.

I think you are right. I always run with the intersection. What's the behavior you would like to see in that case (what value would you like to see in the correlation matrix? I can update it again

Reply:
Yes, I think it needs to be adapted for union, I will take it from here and I will push the code in a moment.